File: config-core/src/main/java/io/gitee/zicai/core/feign/IAppFeign.java
package io.gitee.zicai.core.feign;

import io.gitee.zicai.core.vo.ResultVO;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

public interface IAppFeign {

    String PREFIX = "/app";

    @GetMapping(PREFIX + "/get/{appName}/{profile}")
    ResultVO appGet(@PathVariable("appName") String appName, @PathVariable("profile") String profile);

    @GetMapping(PREFIX + "/add/{appName}/{profile}")
    ResultVO appAdd(@PathVariable("appName") String appName, @PathVariable("profile") String profile);

    @GetMapping(PREFIX + "/del/{id}")
    ResultVO appDel(@PathVariable("id") Long id);

    @GetMapping(PREFIX + "/page/{index}/{size}")
    ResultVO appPage(@PathVariable("index") Integer index, @PathVariable("size") Integer size);
}

File: config-core/src/main/java/io/gitee/zicai/core/feign/IAppFeign.kt
package io.gitee.zicai.core.feign

import io.gitee.zicai.core.vo.ResultVO
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable

interface IAppFeign {

    companion object {
        const val PREFIX = "/app"
    }

    @GetMapping("$PREFIX/get/{appName}/{profile}")
    fun appGet(@PathVariable("appName") appName: String, @PathVariable("profile") profile: String): ResultVO

    @GetMapping("$PREFIX/add/{appName}/{profile}")
    fun appAdd(@PathVariable("appName") appName: String, @PathVariable("profile") profile: String): ResultVO

    @GetMapping("$PREFIX/del/{id}")
    fun appDel(@PathVariable("id") id: Long): ResultVO

    @GetMapping("$PREFIX/page/{index}/{size}")
    fun appPage(@PathVariable("index") index: Int?, @PathVariable("size") size: Int?): ResultVO
}
